app/models/user.rb

```diff
@@ -245,21 +245,10 @@ def cached_following_podcasts_ids
     end
   end
 
-  # handles both old (prefer_language_*) and new (Array of language codes) formats
   def preferred_languages_array
     return @preferred_languages_array if defined?(@preferred_languages_array)
 
-    if language_settings["preferred_languages"].present?
-      @preferred_languages_array = language_settings["preferred_languages"].to_a
-    else
-      languages = []
-      language_settings.each_key do |setting|
-        to_split = language_settings[setting] && setting.include?("prefer_language_")
-        languages << setting.split("prefer_language_")[1] if to_split
-      end
-      @preferred_languages_array = languages
-    end
-    @preferred_languages_array
+    @preferred_languages_array = language_settings["preferred_languages"]
```

Comment:
Not a blocker for this PR, but looking at the pre-existing code, I wonder if we could do `@preferred_languages_array ||= ` here and return the variable if it exists, rather than `return @preferred_languages_array if defined?(@preferred_languages_array)` ?

Reply:
Agreed! My plan is to sweep that up in a future round once we know more about how we're going to use this Array after we scope out the project more. If we add back in some logic, then the existing setup would need to remain/be put back in.